lib/astutils.cpp

```diff
@@ -241,9 +241,14 @@ bool astIsIterator(const Token *tok)
     return tok && tok->valueType() && tok->valueType()->type == ValueType::Type::ITERATOR;
 }
 
-const Library::Container* astIsContainer(const Token* tok) {
+bool astIsContainer(const Token* tok) {
+    return getLibraryContainer(tok) != nullptr && !astIsIterator(tok);
+}
+
+bool astIsUniversalContainer(const Token* tok)
```

Comment:
What does universal container mean? It seems like this function is specific to lifetime analysis, maybe it should be moved to valueflow.cpp. I would also prefer a more descriptive name like `astIsNonStringContainer`.